php-frontend/src/main/java/org/sonar/php/tree/symbols/SymbolVisitor.java

```diff
@@ -498,28 +499,28 @@ public void visitFunctionCall(FunctionCallTree tree) {
     }
     String callee = SourceBuilder.build(tree.callee()).trim();
     if ("compact".equals(callee) || "\\compact".equals(callee)) {
-      visitCompactFunctionCall(tree.arguments());
+      visitCompactFunctionCall(tree.callArguments());
     }
 
-    scan(tree.arguments());
+    scan(tree.callArguments());
```

Comment:
I think we have to be careful here with replacing `scan(tree.arguments())` by `scan(tree.callArguments())`. With the second we will visit more trees, in particular we will visit the possible `name` of a named argument (which is a `NameIdentifierTree`). We wouldn't have visited it before this refactoring.
It is also important for symbol table creation which `NameIdentifierTree` we visit. See `visitNameIdentifier()` in this class. To illustrate the regression, this test fails now, but wasn't before:
```java
  @Test
  public void nested_function() {
    CompilationUnitTree cut = parseSource("<?php \n" +
      "class Foo {" +
      "}\n" +
      "someCall(Foo: $arg);");
    SymbolTableImpl symbolTable = SymbolTableImpl.create(cut);
    assertSymbolUsages(symbolTable, "foo"); // no usages
  }
```
To sum up: I we should replace it with `tree.callArguments().stream().map(CallArgumentTree::value).collect(Collectors.toList())` to avoid regressions. Also check the other one on line `486`

Reply:
Fair point!